### PR TITLE
test: added tests for selectedfilters component

### PR DESCRIPTION
**Pull request by AlexanderDevitsky**
## Related Issue

https://vsf.atlassian.net/browse/M2-436

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import userEvent from '@testing-library/user-event';
+import { render } from '~/test-utils';
+import { RemovableFilter, removableFiltersData } from '~/test-utils/mocks/removableFiltersMock';
+import SelectedFilters from '~/modules/catalog/category/components/filters/FiltersSidebar/SelectedFilters.vue';
+
+describe('SelectedFilters.vue', () => {
+  it('Should not have children if if the component has no data', () => {
+    const { container } = render(SelectedFilters);
+
+    expect(container.querySelector('.selected-filter')).toBeNull();
```
Comment (Frodigo) [inline packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts:10]: Please use here `findByTestId('selected-filter')`

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+  it('Should render selected filters if options are exist', () => {
+    const { getAllByTestId } = render(SelectedFilters, {
+      props: {
+        removableFilters: removableFiltersData as RemovableFilter[],
+      },
+    });
+    const selectedFilers = getAllByTestId('selected-filter');
+
+    selectedFilers.forEach((filter) => {
```
Comment (Frodigo) [inline packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts:21]: Remove this for each, please.

**Review by Frodigo — CHANGES_REQUESTED** [review]

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts (hunk 1 above)
Comment (AlexanderDevitsky) [reply packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts]: findBy query returns a promise if the element is found and throws an error if the element doesn't exist, so async/await construction should be used with it. I decided to use queryByTestId. Perhaps this is better?
Screenshot 2022-05-27 at 13 54 07

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts (hunk 2 above)
Comment (AlexanderDevitsky) [reply packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts]: removed

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts (hunk 1 above)
Comment (Frodigo) [reply packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts]: yes, sorry I meant queryBy, so my mistake. Yes, go ahead with queryBy!

Comment (AlexanderDevitsky) [reply packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts]: done

**Review by Frodigo — DISMISSED** [review]

File: packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    });
+    const removeButton = getByTestId('selected-filter-remove');
+
+    expect(removeButton).toBeDefined();
+
+    await userEvent.click(removeButton);
+
+    expect(emitted().removeFilter).toHaveLength(1);
+  });
+});
```
Comment (bartoszherba) [inline packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts:38]: Can we also test if `yes/no` label is rendered properly?

Comment (AlexanderDevitsky) [reply packages/theme/modules/catalog/category/components/filters/FiltersSidebar/__tests__/SelectedFilters.spec.ts:68]: @bartoszherba added

**Review by Frodigo — APPROVED** [review]

**Pull request merged**